File: mwc/romana/relic/b/lib/libc/string/i8086/memcpy.m
//////////
/ i8086 C string library.
/ memcpy()
/ ANSI 4.11.2.1.
//////////

//////////
/ char *
/ memcpy(To, From, Count)
/ char *To, *From;
/ int Count;
/
/ Copy Count bytes from From to To.
//////////

#include <larges.h>

To	=	LEFTARG
From	=	To+DPL
Count	=	From+DPL

	Enter(memcpy_)
	mov	cx, Count(bp)	/ Count to CX
	Lds	si, From(bp)	/ From address to DS:SI
	Les	di, To(bp)	/ To address to ES:DI
	mov	ax, di		/ Result to AX
#if	LARGEDATA
	mov	dx, es		/ or DX:AX
#endif
	cld
	rep
	movsb
	Leave

File: mwc/romana/relic/b/lib/libc/string/memcpy.c
/*
 * memcpy.c
 * ANSI 4.11.2.1.
 * Copy memory.
 */

#include <string.h>

char *memcpy(s1, s2, n) char *s1; register char *s2; register size_t n;
{
	register char *cp;

	for (cp = s1; n--; )
		*cp++ = *s2++;
	return (s1);
}
